metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
`parseOptional` is the only one called by the parseBolt.  This just allows the parser creator to choose between whether they want to use java 8's optional to indicate no data is returned or just return null.  For a similar reason to why people had issue changing the interface during this PR, it wasn't ideal to change the interface when we moved to java 8 back when this was introduced.